data_structures/src/chain.rs

```diff
@@ -2913,10 +2913,46 @@ impl ReputationEngine {
         self.ars
             .active_identities()
             .cloned()
-            .sorted_by_key(|&pkh| std::cmp::Reverse((self.trs.get(&pkh), pkh)))
+            .sorted_by(|a, b| compare_reputed_pkh(a, b, &self).reverse())
             .collect()
     }
 }
+/// Compare 2 PublicKeyHashes comparing:
+/// First: reputation
+/// Second: A hash calculated with the PublicKeyHash and the Alpha value
+/// Third: PublicKeyHash
+fn compare_reputed_pkh(
+    a: &PublicKeyHash,
+    b: &PublicKeyHash,
+    rep_eng: &ReputationEngine,
+) -> Ordering {
+    let rep_a = rep_eng.trs().get(a).0;
+    let rep_b = rep_eng.trs().get(b).0;
+
+    rep_a
+        .cmp(&rep_b)
+        .then({
```

Comment:
This way the hash is still computed every time, but the `then_with` method does exactly what you want

```suggestion
        .then_with(|| {
```